packages/html/src/events/visible.rs

```diff
@@ -0,0 +1,325 @@
+use std::fmt::{Display, Formatter};
+
+pub struct VisibleData {
+    inner: Box<dyn HasVisibleData>,
+}
+
+impl<E: HasVisibleData> From<E> for VisibleData {
+    fn from(e: E) -> Self {
+        Self { inner: Box::new(e) }
+    }
+}
+
+impl VisibleData {
+    /// Create a new VisibleData
+    pub fn new(inner: impl HasVisibleData + 'static) -> Self {
+        Self {
+            inner: Box::new(inner),
+        }
+    }
+
+    /// Get the bounds rectangle of the target element
+    pub fn get_bounding_client_rect(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_bounding_client_rect()
+    }
+
+    /// Get the ratio of the intersectionRect to the boundingClientRect
+    pub fn get_intersection_ratio(&self) -> VisibleResult<f64> {
+        self.inner.get_intersection_ratio()
+    }
+
+    /// Get the rect representing the target's visible area
+    pub fn get_intersection_rect(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_intersection_rect()
+    }
+
+    /// Get if the target element intersects with the intersection observer's root
+    pub fn is_intersecting(&self) -> VisibleResult<bool> {
+        self.inner.is_intersecting()
+    }
+
+    /// Get the rect for the intersection observer's root
+    pub fn get_root_bounds(&self) -> VisibleResult<PixelsRect> {
+        self.inner.get_root_bounds()
+    }
+
+    /// Get the element whose intersection with the root changed
+    pub fn get_target(&self) -> VisibleResult<ElementId> {
```

Comment:
`ElementId` isn't very useful in user-facing code. It is mainly used in core/the renderers. We could use `MountedData` here which wraps the `ElementId` or `web_sys::Element` with accessors.

The target here will always be the element that the event listener is attatched to right? If that is true, then it might be better to wait until we add `target() -> MountedData` to all events together.

Reply:
This is also my understanding (cf. https://css-tricks.com/a-few-functional-uses-for-intersection-observer-to-know-when-an-element-is-in-view/). Are you referring to the https://github.com/DioxusLabs/dioxus/issues/2858?

Reply:
> Are you referring to the https://github.com/DioxusLabs/dioxus/issues/2858?
Yes, both `relatedTarget` and `target` need the mounteddata for the element